wrapper/src/main/java/software/amazon/jdbc/plugin/bluegreen/BlueGreenStatusProvider.java

```diff
@@ -509,6 +521,25 @@ protected void updateDnsFlags(BlueGreenRole role, BlueGreenInterimStatus interim
       LOGGER.finest(() -> Messages.get("bgd.greenTopologyChanged", new Object[] {this.bgdId}));
       this.greenTopologyChanged = true;
       this.storeGreenTopologyChangeTime();
+      this.resetMonitors(this.monitorResetOnTopologyCompleted, "- green topology");
+    }
+  }
+
+  protected void resetMonitors(final AtomicBoolean monitorResetCompleted, final String eventName) {
+    if (monitorResetCompleted.compareAndSet(false, true)) {
+
+      final Set<String> blueEndpoints = this.summaryStatus.getRoleByHost().entrySet().stream()
+          .filter(x -> x.getValue() == BlueGreenRole.SOURCE)
+          .map(Entry::getKey)
+          .collect(Collectors.toSet());
+
+      // A special case for efm related monitors.
+      // TODO: remove this code as soon as efm1 monitors are migrated to MonitorService
+      HostMonitorThreadContainer.resetMonitors(blueEndpoints);
+
+      // Notify all monitors to reset if needed.
+      this.servicesContainer.getEventPublisher().publish(new MonitorResetEvent(this.clusterId, blueEndpoints));
+      this.storeMonitorResetTime(eventName);
```

Comment:
Can we also remove the `getInterimStatusHash` method low, it is no longer in use